File: Julia/day2-2.jl
function isvalidpassword(password::AbstractString, char::AbstractChar,
    firstchar::Integer, secondchar::Integer)
    return (password[firstchar] == char) ⊻ (password[secondchar] == char)
end

function processline(line::AbstractString)
    re = r"(\d*)-(\d*) (\w): (\w*)"

    firstchar, secondchar, character, password = match(re, line).captures

    firstchar = parse(Int, firstchar)
    secondchar = parse(Int, secondchar)

    return password, character[1], firstchar, secondchar
end

nvalid = 0
open("Recursos/day2-input.txt") do io
    for line in eachline(io)
        password, char, firstchar, secondchar = processline(line)

        global nvalid += isvalidpassword(password, char, firstchar, secondchar)
    end
end

println(nvalid)

File: Python/day2-2.py
import re


def isvalidpassword(password: str, char: str, firstchar: int, secondchar: int):
    return (password[firstchar-1] == char) ^ (password[secondchar-1] == char)


def processline(line: str):
    template = "(\d*)-(\d*) (\w): (\w*)"

    firstchar, secondchar, character, password = re.match(
        template, line).groups()

    firstchar = int(firstchar)
    secondchar = int(secondchar)

    return password, character, firstchar, secondchar


nvalid = 0
with open("Recursos/day2-input.txt") as io:
    for line in io:
        nvalid += isvalidpassword(*processline(line))

print(nvalid)
